scalafix-core/shared/src/main/scala/scalafix/util/TokenList.scala

```diff
@@ -8,40 +8,29 @@ import scala.meta.tokens.Tokens
 /** Helper to traverse tokens as a doubly linked list.  */
 final class TokenList private (tokens: Tokens) {
   def trailing(token: Token): SeqView[Token, IndexedSeq[Token]] =
-    tokens.view(tok2idx(token), tokens.length).drop(1)
+    tokens.view(tok2idx(token) + 1, tokens.length)
+
   def leading(token: Token): SeqView[Token, IndexedSeq[Token]] =
-    tokens.view(0, tok2idx(token)).drop(1).reverse
+    tokens.view(0, tok2idx(token)).reverse
+
   private[this] val tok2idx = {
     val map = Map.newBuilder[Token, Int]
     var i = 0
     tokens.foreach { tok =>
       map += (tok -> i)
       i += 1
     }
-    map.result()
+    map
+      .result()
+      .withDefault(t =>
+        throw new NoSuchElementException(s"token not found: $t"))
   }
 
-  def find(start: Token)(f: Token => Boolean): Option[Token] = {
-    def loop(curr: Token): Option[Token] = {
-      if (f(curr)) Option(curr)
-      else {
-        val iter = next(curr)
-        if (iter == curr) None // reached EOF
-        else loop(iter)
-      }
-    }
-    loop(next(start))
-  }
+  def find(start: Token)(p: Token => Boolean): Option[Token] =
+    tokens.drop(tok2idx(start)).find(p)
 
-  def slice(from: Token, to: Token): Seq[Token] = {
-    val builder = Seq.newBuilder[Token]
-    var curr = next(from)
-    while (curr.start < to.start) {
-      builder += curr
-      curr = next(curr)
-    }
-    builder.result()
-  }
+  def slice(from: Token, to: Token): SeqView[Token, IndexedSeq[Token]] =
```

Comment:
Is this signature change necessary? I think we can keep the signature to avoid a binary breaking change.

Reply:
In fact it is not. Given that the method now returns a view, I wanted to make this fact explicit as well as have the return type consistent the `leading*`/`trailing*` methods. It wouldn't cause compilation errors but indeed it would break binary compatibility. I'm gonna change it back to `Seq` to avoid that.